### PR TITLE
fix CRD / Codegen

### PR DESCRIPTION
**Pull request by squeed**
- make sure codegen is vendored in 
- bump controller-tools
- add in the manually generated annotations

### CODE REVIEW

**Comment by danielmellado** [comment]
Hi @squeed it does seem like the CI is complaining about the magnet, mind taking a look before I cherry-pick this? Thanks!

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@ if  ! ( command -v controller-gen > /dev/null ); then
   fi
 fi
 
-"${SCRIPT_ROOT}/vendor/k8s.io/code-generator/generate-groups.sh" deepcopy \
+bash "${SCRIPT_ROOT}/vendor/k8s.io/code-generator/generate-groups.sh" deepcopy \
```
Comment (danwinship) [inline hack/update-codegen.sh:21]: why?

File: hack/verify-style.sh [hunk 2]
```diff
@@ -23,8 +23,8 @@ trap 'rc=$?' ERR
 # Go to the root of the repo
 cd "$(git rev-parse --show-cdup)"
 
-GOFILES=$(find . -path ./vendor -prune -o -name '*.go' | grep -v vendor | grep -v pkg/operator/assets)
-GOPKGS=$(go list ./... | grep -v '/vendor/' | grep -v '/generated/' | grep -v pkg/operator/assets)
+GOFILES=$(find . -path ./vendor -prune -o -name '*.go' | grep -v vendor)
```
Comment (danwinship) [inline hack/verify-style.sh:26]: fwiw

    GOFILES=$(find . -path ./vendor -prune -o -name '*.go' -print)

works too. (ie, explicitly telling it to print filenames that match the `-name` predicate overrides the default behavior of printing filenames that match any predicate)

File: hack/verify-style.sh [hunk 3]
```diff
@@ -23,8 +23,8 @@ trap 'rc=$?' ERR
...
+GOPKGS=$(go list ./cmd... ./pkg/...)
```
Comment (danwinship) [inline hack/verify-style.sh:27]: the old version was written before `go` knew about `vendor`. Just

    GOPKGS=$(go list ./...)

will work too. (And picks up the `version` subdir, which I'm not sure if you had meant to omit)

File: hack/update-codegen.sh [hunk 4]
```diff
@@ -6,14 +6,12 @@ set -o pipefail
 
 SCRIPT_ROOT="$(dirname "${BASH_SOURCE[0]}")/.."
 
-if  ! ( command -v controller-gen > /dev/null ); then
-  # Need to take an unreleased version of controller-tools
-  # because of controller-tools bug 302
-  echo "controller-gen not found, installing sigs.k8s.io/controller-tools@83f6193..."
+if ( ! ( command -v controller-gen > /dev/null )  || test "$(controller-gen --version)" != "Version: v0.4.0" ); then
+  echo "controller-gen not found or out-of-date, installing sigs.k8s.io/controller-tools@v0.4.0..."
```
Comment (danwinship) [inline hack/update-codegen.sh:10]: if [[ $(controller-gen --version 2>/dev/null) != "Version: v0.4.0" ]]; then

?

also, "incorrect version", not "out of date", though do we really want to be blocking people from using v0.4.1+? Are we assuming API incompatibilities?

File: manifests/0000_70_cluster-network-operator_01_pki_crd.yaml [hunk 5]
```diff
@@ -1,12 +1,12 @@
-
----
-apiVersion: apiextensions.k8s.io/v1beta1
+# This file is automatically generated. DO NOT EDIT
+apiVersion: apiextensions.k8s.io/v1
```
Comment (damemi) [inline manifests/0000_70_cluster-network-operator_01_pki_crd.yaml:2]: Not sure who to ask, but do you also want to update the CRD in https://github.com/openshift/cluster-network-operator/blob/master/bindata/network/openshift-sdn/001-crd.yaml#L2 to `v1` as well? (from https://bugzilla.redhat.com/show_bug.cgi?id=1881636#c4, this BZ was originally for upgrading all the CRDs in openshift/api to v1)

If so, you may want to do it here and link this PR to that bugzilla

File: hack/update-codegen.sh (hunk 1 above)
Comment (squeed) [reply hack/update-codegen.sh:21]: because the shell script isn't executable.

File: hack/update-codegen.sh (hunk 4 above)
Comment (squeed) [reply hack/update-codegen.sh:10]: I hate it when I do codegen (in any of its forms) and the output changes in ways I don't understand. So, I prefer to pin versions.

File: hack/verify-style.sh (hunk 2 above)
Comment (squeed) [reply hack/verify-style.sh:26]: all of this is silly - filed SDN-1342 to switch to golangci-lint.

File: hack/verify-style.sh (hunk 3 above)
Comment (squeed) [reply hack/verify-style.sh:27]: We need to skip `dependencymagnet`, hence the changes here. (And version will go away soonish)

File: manifests/0000_70_cluster-network-operator_01_pki_crd.yaml (hunk 5 above)
Comment (squeed) [reply manifests/0000_70_cluster-network-operator_01_pki_crd.yaml:2]: This is just to fix codegen; let's make it a separate PR.

**Comment by squeed** [comment]
This isn't going to pass until https://github.com/openshift/release/pull/13489 merges.

**Comment by squeed** [comment]
hey @tssurya, I know this is blocking you - want to review it?

File: hack/update-codegen.sh (hunk 4 above)
Comment (danielmellado) [reply hack/update-codegen.sh:10]: What would be changed without this pinning exactly?

**Comment by squeed** [comment]
/hold
something's broken.

File: hack/update-codegen.sh (hunk 4 above)
Comment (squeed) [reply hack/update-codegen.sh:10]: @danielmellado if we don't pin a specific version, then people who run `hack/update-codegen.sh` might get different output depending on the version of the binary installed on their local machine.

**Comment by squeed** [comment]
Test failures are spurious. I think we can merge this. Someone want to lgtm?

**Comment by tssurya** [comment]
> hey @tssurya, I know this is blocking you - want to review it?

oops, too late to the party, I need to get better with doing reviews diligently but thanks for doing this, I'll rebase my patch on this.

**Pull request merged**